pkg/zabbix_api.go

```diff
@@ -112,16 +112,22 @@ func (ds *ZabbixDatasource) BuildResponse(result *simplejson.Json) (*datasource.
 func (ds *ZabbixDatasource) ZabbixRequest(ctx context.Context, dsInfo *datasource.DatasourceInfo, method string, params *simplejson.Json) (*simplejson.Json, error) {
 	zabbixUrl := dsInfo.GetUrl()
 
-	// Authenticate first
-	if zabbixAuth == "" {
+	var result *simplejson.Json
+	var err error
+
+	for attempt := 0; attempt <= 3; attempt++ {
+		// Authenticate
 		auth, err := ds.loginWithDs(ctx, dsInfo)
 		if err != nil {
 			return nil, err
 		}
 		zabbixAuth = auth
```

Comment:
This appears to be stored at the package level. If I understand the backend API correctly, we may be handling requests from multiple different Zabbix datasources during this runtime though. @alexanderzobnin

Reply:
@alecxvs yes, you're right. This should be stored in datasource-specific object. I think we should create a cache instance for each datasource.

Reply:
I think we can leave it as is for now, but I'll create an issue for moving things into datasource-specific cache.

Reply:
@alecxvs I created an issue #812